src/PointwiseFunctions/AnalyticSolutions/Poisson/Moustache.cpp

```diff
@@ -0,0 +1,116 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticSolutions/Poisson/Moustache.hpp"
+
+#include <algorithm>
+#include <array>
+#include <pup.h>  // IWYU pragma: keep
+
+#include "DataStructures/DataVector.hpp"     // IWYU pragma: keep
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/MakeWithValue.hpp"
+// IWYU pragma: no_forward_declare Tensor
+
+namespace Poisson {
+namespace Solutions {
+
+namespace {
+
+template <size_t Dim>
+tnsr::I<DataVector, Dim> auxiliary_field(
```

Comment:
Why not just overload the function instead of making it a template?

Reply:
Is that possible with the argument `x` having the template parameter `Dim`? It's a different implementation for both dimensionalities. Not sure what you are suggesting.

Reply:
Sorry, what I meant was getting rid of the forward declaration and using:

```
tnsr::I<DataVector, 1> auxiliary_field(
    const tnsr::I<DataVector, 1>& x,
    const DataVector& precomputed_norm_square) noexcept {
  ..
}

tnsr::I<DataVector, 2> auxiliary_field(
    const tnsr::I<DataVector, 2>& x,
    const DataVector& precomputed_norm_square) noexcept {
  ..
}
```

Does that make sense?

Reply:
Alright got it, thanks